### DISCUSSION TITLE
SolarEdge support for charging via Clipping only

### DISCUSSION THREAD

**Discussion by LupoLoopy** [springfall2008/batpred#1126] [Q&A]
Hi

Does BatPred have any support for charging via Clipped energy only? 

I have a 3.67kW inverter (pretty standard in the UK) so there are often days where I'd be happy for the electricity I capture to be preferentially exported if the projected solar generation is going to cover my needs, allowing for up to 5kW of clipped energy. I'm not in a rush to charge the battery under those circumstances, as long as it is going to be full by sundown. 

The current approach of racing to fill the battery leaves a lot of potential energy on the proverbial table

Thanks!

**Comment by Noodleyman** [comment]
What's your inverter model number? it's an odd size inverter. I asusme you mean export cap limit? With SE kit you can oversize up to 200% which I would recommend taking advantage of where viable :). even with the low export limit you can get more panels if you have the space for them. 

also in the UK and have a 3.68kWh export limit (booo!) I solved this with my setup by overriding PredBat with automations, however my setup is quite large. 8kW inverter with 15kWp.
I use Octopus Flux, so have a charge window of 2-5AM.

I currently use automations to disable read only mode at  1:45AM then enable read only mode at 5:15AM if the solar forecast is going to be more than 30kWh for the day. I also set a helper to track that Predbat has been overriden for the day. At this point everything is in Max Self Consumption mode. 

I've another automation which tracks when we're overrideen. When my battery is > 15% and my PV generation is covering usage and has been for at least 3 minutes I set Charge from Clipped mode to boost exports. I use another helper which resets after 3 mins to prevent flip/flop between modes. thus when usage has been below generation flip to charge from clipped, and when usage is over generation for 3 mins flip to max self consumption mode. 

At 2PM the override of the system is removed, and read only mode disabled, thus if PredBat needs to do any charging it has the option to do so before the 4PM peak rates kick in. 

This system has been working really well for my setup and is maxing my constrained exports. You could do something similar and set an export rate limit to suit your own needs (as long as it's within DNO requirements)  

I also found that Predbat holds seem to not work correctly with SE kit, or at least my own kit. I've again used automations to capture those conditions and workaround them where viable. IE setting the charge and discharge rate limits to 0 always trigger them to revert back to 5000 because 0 isn't valid. so when it sets 0, I set to 1 instead which does work.

**Comment by LupoLoopy** [comment]
Have you considered logging the charge and discharge rate number = 0 fault as an issue (unless you already have)?

Good idea on the automation. I'm giving it a go now. I'll need to work out how early I can forcibly stop charging my battery in order to maximise my clipping yield. Also - at what point is it REALLY worth it, as more time cycling the battery may make me more money than the clipping losses, below a certain threshold. 

I can work those out though, once I have the principle of the thing setup.

**Comment by Noodleyman** [comment]
Morning,

The 0 to 1 issue has been discuseed, but I seem to be the only person (I know of) facing this issue, so I didn't raise it in the end. If it's still happening when I get nearer to winter and the dynamic features I want to be in play are still not quite right then I'll dig into it further. 

If it would be of use to you, I'll happily share my yaml for my automations, would you like the code to use as an example? The only thing I want to do but have not done yet is move all the config out of yaml and into settings which can be controlled via the UI.

I've now also got a backup interface so am having the reserve battery set based upon weather warnings in our area. this is also working well for us. I've just started exploring powering down stuff around the home via automations when were in reserve power mode (IE, power is out) to keep the vital stuff working. It's a work in progress.

**Comment by cyberkryten** [comment]
I'd like to be able to charge via clipping too: 3.68kW export limit, 8kW Inverter with 10kW (E/W) panels.

I've been playing around with  "Best SOC Max" to stop Predbat filling the battery overnight on IOG @7.5p/kWh when I think I could get it 'for free' from clipped power.  I'm not very good at it though as battery was full at 08:30 today!

Does anyone know if Predbat has this functionality with other inverters?  If it is just adding the "Charge from Clipped Solar Power" control then I'll have a dig through and see if I can figure out what's needed to add that state for the SolarEdge inverter.

With regards to the Discharge=0 issue, I have various problems with SE including that value resetting to 11400 every night, despite it being an 8kW inverter but the setting to zero does actually work on mine.

**Reply by Noodleyman** [reply]
I've also noted the 11400 issue, but have entirely ignored it as although the value reports 11400 in reality the kit won't exceed 5000.I am not sure why it does it and think it's an oddity with the modbus data. Even when it reports 11400 the kit itself won't exceed the hardware limits of 5000.

will reply to your other points with some thoughts a little later when I get some spare time :)

**Reply by cyberkryten** [reply]
My 8kW inverter will peak its output at 10kW in Discharge mode - both when in Remote Control mode but also when just using the SolarEdge app in Manual Time of Use Discharge mode.  Not sure if that value would confuse PredBat into thinking the Inverter can actually output way more and mess up the calculations?

My issue is that DNO require a limit of 3.68kW export with max output of 7.36kW which is set, but I can see it hitting the full 10 on the Smart Meter monitoring 🤷 

In Remote Control mode I can have an automation that resets the Discharge Limit to 7360 when it detects it has gone higher (which is does every evening around 23:30 when the inverter seems to reset), but in 'App only' mode, there's no way to reduce it!

**Reply by Noodleyman** [reply]
From your earlier message, I think the feature you are asking about in PredBat is the switch **switch.predbat_set_charge_low_power**

What this does (I believe) is rate limit the charge of the battery to slow charge it, thus making more power available for export or self consumption. I think this was added as a way to get a charge from clipped method availble for other systems that perhaps don't have the option. But, if you have the SolarEdge kit, you already have a clipping mode you just need to enable as it's fully supported by SE.

The mode is "Charge from Clipped Solar Power"
![image](https://github.com/springfall2008/batpred/assets/10614018/89a19e41-9f36-44f2-888a-7ffefacbc553)

This mode will firstly cover your live consumption, and then prioritise exports. Once your exporting at the configured limit the remainder goes into the battery. If you find that the export it too much and your battery isn't getting charged, perhaps the feature above via the switch might be useful, or perhaps you can lower your export limit based on some formulas which uses hours until peak rate, kWh required to fill battery etc. I do something similar to toggle max self consumption mode and charge from clipped based upon whatever the consumption needs are. IE, when the home wants more power than the PV can generate for more than a couple of minutes, flip from charge from clipped mode to max self consumption so it will use the battery rather than grid, then flip back to clipped mode when it can. 

The DNO limit seems weird. I get the export limit, but a max output limit? That doesn't seem correct. as long as it's a DC coupled battery, they don't usually care about the ouput from inverter to your home as it doesn't touch the grid iun the same way an AC battery does.  Are you sure that DNO restriction is correct? I also have a 3.68kWh export limit, but I've the SE8000H (8kW) inveter, 15kWP (36 panels), but zero limits with regards to output, and originally the DNO tried to deny me solar all together as our grid is naff (very rural, end of power line).  

The storage charge/discharge you are seeing relates to the power into and out of the battery to/from inverter or grid, so this won't help you limit the output of the inverter itself to the home.

**Reply by cyberkryten** [reply]
I know SE has that as a setting, but I can't find a way to get Predbat to use it?  Is there a setting I've missed somewhere, because if there is, I can't find it.  I also can't find anything referencing it in the code.  I guess I could go back to trying to do my own estimates/calculations, but for days when generation isn't as high, Predbat does a way better job than my calcs were managing.

It makes sense to run in that mode on high generation days because of the export restriction, but by default Predbat seems intent on filling the batteries, so I'm trying to figure out a way to keep using it by tweaking things.

I had similar issues with DNO which is why I ended up with a Hybrid system, the DNO def required 7.36kW max inverter size (the 8kW is allowed as long as output is limited in software).

**Reply by Noodleyman** [reply]
Ah I see. There isn't a way today to get predbat to use it unfortunately that I am aware of. I wish it did though. I think there is an enhancement requst somewhere for it (or I am recalling wrong and one needs to be raised). How are you setting the output limit? I don't recall that being an option anywhere in SetApp. Those settings you see via Home Assistant for your inverter relate to the battery output/input only.  There is a rate limit for taking from the grid when charging the battery, "AC Charge Limit", is this the setting you are enforced to use? I'm intreged as I didn't think you could lower the output from the inverter

**Reply by gcoan** [reply]
> From your earlier message, I think the feature you are asking about in PredBat is the switch switch.predbat_set_charge_low_power
> 
> What this does (I believe) is rate limit the charge of the battery to slow charge it, thus making more power available for export or self consumption.

Unfortunately the set_charge_Low_power switch does limit the charge rate of the battery, but it's only used (at present) when the battery is grid charging (e.g. overnight) and plans the charging so the battery will be full by the end of the charging period.
e.g. charging planned from 1-6am, rather than charge at full rate for 2 hours then idle for 4, it reduces the charge rate to 2/6 of full rate, thus keeping the battery & inverter warm throughout rather than hot then cold.

So this doesn't help with minimising clipping, sorry.

There are some feature requests for @springfall2008 to introduce clipping logic, but it's not there yet.

At the moment I'm getting so much solar (and agile rates are high overnight) that I've been leaving predbat with read-only mode turned on, and letting the battery charge via solar.  I have manually reduced my battery charging rate right down so it charges through the day rather than being fully charged at 10am.  That's my solution, not ideal, and not using predbat unfortunately at the moment.  Once agile rates drop and its worth charging overnight I will turn it back on again

**Reply by Noodleyman** [reply]
Ah, thank you @gcoan , I had totally got that wrong then as I thought it worked differently. the docs don't say that setting only applies to grid charging.  can agree that I also love this time of year for generation! but wish I had a larger export limit (won't happen). even with charging from clipping were charged by around 11AM most days, although I am a power hungry user due to a lot of IT equipment so a full battery doesn't last that long. Waiting on the next gen batteries to come along from SE to see if they will have a cheaper pricepoint than the energy banks which are imo just too costly to add more.
